src/submission/validation-clinical/followUp.ts

```diff
@@ -24,13 +24,82 @@ import {
 	SubmissionValidationOutput,
 	DonorVitalStatusValues,
 } from '../submission-entities';
-import { ClinicalEntitySchemaNames, FollowupFieldsEnum } from '../../common-model/entities';
+import {
+	ClinicalEntitySchemaNames,
+	DonorFieldsEnum,
+	FollowupFieldsEnum,
+} from '../../common-model/entities';
 import { DeepReadonly } from 'deep-freeze';
 import { Donor, Treatment } from '../../clinical/clinical-entities';
 import _ from 'lodash';
 import { getClinicalEntitiesFromDonorBySchemaName } from '../../common-model/functions';
 import { getEntitySubmitterIdFieldName } from '../../common-model/functions';
 import * as utils from './utils';
+import { checkForExceptions } from '../exceptions/exceptions';
+
+const checkDonorTimeConflict = (
+	donorDataToValidateWith: { [k: string]: any },
+	followUpRecord: DeepReadonly<SubmittedClinicalRecord>,
+	errors: SubmissionValidationError[],
+) => {
+	if (
+		donorDataToValidateWith.donorVitalStatus === DonorVitalStatusValues.deceased &&
+		donorDataToValidateWith.donorSurvivalTime <
+			followUpRecord[FollowupFieldsEnum.interval_of_followup]
+	) {
+		errors.push(
+			utils.buildSubmissionError(
+				followUpRecord,
+				DataValidationErrors.FOLLOW_UP_DONOR_TIME_CONFLICT,
+				FollowupFieldsEnum.interval_of_followup,
+				{},
+			),
+		);
+	}
+};
+
+function checkTreatmentTimeConflict(
+	followUpRecord: DeepReadonly<SubmittedClinicalRecord>,
+	treatment: DeepReadonly<Treatment>,
+	errors: SubmissionValidationError[],
+) {
+	// A follow up may or may not be associated with treatment
+	if (treatment == undefined) return;
+
+	if (
+		followUpRecord.interval_of_followup &&
+		treatment.clinicalInfo &&
+		treatment.clinicalInfo.treatment_start_interval &&
```

Comment:
Potential dedupe with `treatment.clinicalInfo?.treatment_start_interval`

Reply:
Will leave as is for now, lots of cases to cleanup with many conditions relying on this check (not just line 72)